src/app/EventManagement.cpp

```diff
@@ -462,15 +481,14 @@ CHIP_ERROR EventManagement::CopyAndAdjustDeltaTime(const TLVReader & aReader, si
 CHIP_ERROR EventManagement::LogEvent(EventLoggingDelegate * apDelegate, EventOptions & aEventOptions, EventNumber & aEventNumber)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
-    CriticalSectionEnter();
+    EventLocked lock;
 
     VerifyOrExit(mState != EventManagementStates::Shutdown, err = CHIP_ERROR_INCORRECT_STATE);
     VerifyOrExit(aEventOptions.mpEventSchema != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
 
     err = LogEventPrivate(apDelegate, aEventOptions, aEventNumber);
 
 exit:
-    CriticalSectionExit();
```

Comment:
Do we want to be holding the lock when we call `ChipLogFunctError`?  Might be better to have a narrower scope for the lock here.

Reply:
Agree, done, thanks

Reply:
This is now back to holding the lock through the `ChipLogFunctError` call.

With an RAII lock, the duration of locking is controlled by scope.  Here, scope can be explicitly introduced with braces:

```
CHIP_ERROR EventManagement::LogEvent(EventLoggingDelegate * apDelegate, EventOptions & aEventOptions, EventNumber & aEventNumber)
{
    CHIP_ERROR err = CHIP_NO_ERROR;
    {
        LockEvents lock;

        VerifyOrExit(mState != EventManagementStates::Shutdown, err = CHIP_ERROR_INCORRECT_STATE);
        VerifyOrExit(aEventOptions.mpEventSchema != nullptr, err = CHIP_ERROR_INCORRECT_STATE);

        err = LogEventPrivate(apDelegate, aEventOptions, aEventNumber);
    } // lock is released here by implicit destruction

exit:
    ChipLogFunctError(err);
    return err;
}
```

Is there a reason this can't work?  Does the `goto` in `VerifyOrExit` somehow prevent it?

Reply:
Done, thanks